docs/ai/conceptual/understanding-tokens.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Understanding tokens"
+description: "Understand how large language models (LLMs) use tokens to analyze semantic relationships and generate natural language outputs"
+author: haywoodsloan
+ms.topic: concept-article
+ms.date: 04/12/2024
+
+#customer intent: As a .NET developer, I want understand how large language models (LLMs) use tokens so I can add semantic analysis and text generation capabilities to my .NET projects.
+
+---
+
+# Understanding tokens
+
+Tokens are words, character sets, or combinations of words and punctuation that are used by large language models (LLMs) to decompose text into. Tokenization is the first step in training. The LLM analyzes the semantic relationships between tokens, such as how commonly they're used together or whether they're used in similar contexts. After training, the LLM uses those patterns and relationships to generate a sequence of output tokens based on the input sequence.
+
+## Turning text into tokens
+
+The set of unique tokens that an LLM is trained on is known as its _vocabulary_.
+
+For example, consider the following sentence:
+
+> I heard a dog bark loudly at a cat
+
+This text could be tokenized as:
+
+- I
+- heard
+- a
+- dog
+- bark
+- loudly
+- at
+- a
+- cat
+
+The model will add only the first occurrence of the "a" token to it's vocabulary. All other occurrences can be represented by that first token.
+
+By having a sufficiently large set of training text, tokenization can compile a vocabulary of many thousands of tokens.
+
+## Common tokenization methods
+
+The specific tokenization method varies by LLM. Common tokenization methods include:
+
+- **Word** tokenization (text is split into individual words based on a delimiter)
+- **Character** tokenization (text is split into individual characters)
+- **Subword** tokenization (text is split into partial words or character sets)
+
+For example, the GPT models, developed by OpenAI, use _Byte-Pair Encoding_ (BPE) which is a type of subword tokenization. OpenAI provides [a tool to visualize how text will be tokenized](https://platform.openai.com/tokenizer).
```

Comment:
```suggestion
For example, the GPT models, developed by OpenAI, use a type of subword tokenization that's known as _Byte-Pair Encoding_ (BPE). OpenAI provides [a tool to visualize how text will be tokenized](https://platform.openai.com/tokenizer).
```
This is another change for localization. "Which" doesn't always translate well. We try to stick with "that" -- or rewrite.